### PR TITLE
UX improvements change status dialog

### LINKED ISSUE #23411: Setting empty custom status raises an error
### Steps to reproduce
1. Open the "set status" dialog
2. Pick an emoji in the custom status section
3. Leave the "What's your status" field empty
4. Click "Set status" to save

### Expected behaviour

Status cleared.

### Actual behaviour

Server error.

It appears that the empty string message gets converted to "null", which the server doesn't accept:
image

### Server configuration

Happens with both Nextcloud 21.0.0 alpha (git master 349da3395a2a5e56fc0cdb77ae8e67182842d0bd) and also da2c88ef28 which is the merge commit from https://github.com/nextcloud/server/pull/23108

### Browsers

Tested with Firefox 80.0 and Chromium 85.0.4183.121

### PR DESCRIPTION
**Pull request by PVince81**
### Description

Focus on the custom message field after picking an emoji.
Enter key in the custom message field saves the status.
~~Disable the save button when no custom message is entered (would result
in a server error).~~
Disable save buttons while saving is in progress.

Fixes setting an empty message for clearing when not touching the field (https://github.com/nextcloud/server/issues/23411)

### Motivation
~~
This was mostly because I noticed that entering no custom message triggered an error. I didn't manage to make the server accept empty messages (and even then who knows what side effects...), so I went with prevent empty status message.
And as a bonus added the enter key + autofocus.
~~

### Issues

Partial fix for https://github.com/nextcloud/server/issues/23276 by introducing "save by enter key"
Fixes https://github.com/nextcloud/server/issues/23411

### CODE REVIEW

**Comment by PVince81** [comment]
Regarding compiled assets, I had to guess by looking at other PRs.
I've updated the README to make it clearer: https://github.com/nextcloud/server/pull/23306

File: apps/user_status/src/components/SetStatusModal.vue [hunk 1]
```diff
@@ -49,19 +49,21 @@
...
 			<ClearAtSelect
 				:clear-at="clearAt"
 				@selectClearAt="setClearAt" />
 			<div class="status-buttons">
-				<button class="status-buttons__select" @click="clearStatus">
+				<button class="status-buttons__select" :disabled="isSavingStatus" @click="clearStatus">
 					{{ $t('user_status', 'Clear status message') }}
 				</button>
-				<button class="status-buttons__primary primary" @click="saveStatus">
+				<button class="status-buttons__primary primary" :disabled="isSavingStatus || !message" @click="saveStatus">
```
Comment (nickvergessen) [inline apps/user_status/src/components/SetStatusModal.vue:66]:
```suggestion
				<button class="status-buttons__primary primary" :disabled="isSavingStatus" @click="saveStatus">
```

Setting to empty clears the status just fine, might be less confusing

Comment (PVince81) [reply apps/user_status/src/components/SetStatusModal.vue:66]: It didn't in my instance and caused a server error as it was sending a status icon + empty message.
I personally wanted to set a status without message at that time, this is how I found the issue. Another user might think likewise. So if we fix it to clear the status on empty, the icon would not be set and would not match the user's expectation here.

Comment (nickvergessen) [reply apps/user_status/src/components/SetStatusModal.vue:66]: https://github.com/nextcloud/server/pull/23108

Comment (PVince81) [reply apps/user_status/src/components/SetStatusModal.vue:66]: ok, then it's either a regression or I found an action path that doesn't work:
image

I'll split this into a separate PR and keep this PR here for just the enter key improvements

**Comment by PVince81** [comment]
I've now removed the code related to empty messages.

The disabled attribute is still there but only for disabling the button when saving as we do in other places.

Adjusted and squashed.

**Comment by PVince81** [comment]
I've raised https://github.com/nextcloud/server/issues/23411 for the empty status issue

**Comment by PVince81** [comment]
I did rebuild and check in the assets. Strange.

After rerunning `make dev-setup` from scratch the build marked those files again as modified.
Anyway, I've pushed them now.

**Comment by PVince81** [comment]
ouch, pushed on wrong branch... fixing...

**Comment by PVince81** [comment]
okay, I decided to push the empty message fix here as well to avoid needless future conflicts.

Fixes https://github.com/nextcloud/server/issues/23411

**Review by jancborchardt — APPROVED** [review]
Good stuff design-wise, great work @PVince81! :)

**Comment by PVince81** [comment]
there was a weird unrelated error:
```
1) Test\Files\Cache\ScannerTest::testETagRecreation
Failed asserting that null is of type "string".
```

build restarted

**Comment by PVince81** [comment]
@nickvergessen CI green, can you review teh codez ?

**Comment by rullzer** [comment]
@PVince81 could you rebase so we can get this in?

**Review by ChristophWurst — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Comment by PVince81** [comment]
Rebased and rebuilt the module in question, as it was the only conflict.

**Pull request merged**